packages/compartment-mapper/src/node-modules.js

```diff
@@ -279,6 +293,15 @@ const graphPackage = async (
     parsers: inferParsers(packageDescriptor, packageLocation),
   });
 
+  await Promise.all(
+    values(result.exports).map(async item => {
+      const descriptor = await readDescriptorUpwards(item);
```

Comment:
I presume the performance of this is okay because `readDescriptor` is memoized?

Reply:
Yes. I'd gladly avoid the whole thing altogether, but that's the best we can do.
As for optimizing it, I've been thinking of priming the fs.stat with a lookup table of paths that exist to avoid expensive errors on reading paths that don't exist. But that'd essentially be a node contribution or a package wrapping fs